### PR TITLE
Python 3: Drop use of flush on ResponseWriter in a couple of files

### PR DESCRIPTION
**Pull request by ziransun**
This patch only drops the use of response.write.flush() in a couple of
files as they directly affect test results in Python 3.

Detailed discussions on dropping use of flush on ResponseWriter can be
found at https://github.com/web-platform-tests/wpt/pull/24617
(Also see #24719 )

### CODE REVIEW

File: fetch/range/resources/long-wav.py
Comment (Hexcles) [inline fetch/range/resources/long-wav.py:121]: This seems wrong (it's altering the content of the stream). We should test the return value of `response.writer.write(to_send)` at the end.

File: html/semantics/embedded-content/the-iframe-element/support/download_stash.py
Comment (Hexcles) [inline html/semantics/embedded-content/the-iframe-element/support/download_stash.py:26]: This one is a bit more problematic. We need to change `write_content` (and correspondingly `write_raw_content` and `write_content_file`) to surface the return value so that we can check here, to avoid adding more stuff to the response than the test author wanted.

**Review by Hexcles — CHANGES_REQUESTED** [review]

File: fetch/range/resources/long-wav.py
Comment (ziransun) [reply fetch/range/resources/long-wav.py]: Sorry, I misread #24617. Code updated :)

File: html/semantics/embedded-content/the-iframe-element/support/download_stash.py
Comment (ziransun) [reply html/semantics/embedded-content/the-iframe-element/support/download_stash.py]: Code updated. Thank you!

File: fetch/range/resources/long-wav.py
Comment (Hexcles) [inline fetch/range/resources/long-wav.py:125]: Note that this slightly changes the semantics (in the edge case when the first `write` outside of the `while` loop fails), but it should be fine I think (if the first `write` fails then we'd now call a second `write` that'd fail immediately). Just in case, have you verified the test resutls?

**Review by Hexcles — COMMENTED** [review]
This now largely LGTM.

However, based on what we learned from https://crbug.com/1106861#c18, we should split the change in `wptserve` out into a separate PR, land it first, roll it to Chromium (done by someone on our team), and then finally land the changes in the handlers. Sorry for the extra trouble.

**Comment by ziransun** [comment]
> we should split the change in `wptserve` out into a separate PR

https://github.com/web-platform-tests/wpt/pull/24716

File: fetch/range/resources/long-wav.py
Comment (ziransun) [reply fetch/range/resources/long-wav.py:125]: Yes, I verified the test results locally.

**Review by Hexcles — APPROVED** [review]
LGTM (**not** merging as it's blocked by #24716 )

**Comment by Hexcles** [comment]
@ziransun I've merged #24716 to master; could you rebase this PR and check the results again?

Also, @LukeZielinski could you post here once you've done the weekly roll of WPT tools in Chromium? After that, it should be safe to merge this PR.

**Comment by LukeZielinski** [comment]
Chromium roll of tools has now landed.

**Comment by Hexcles** [comment]
Thanks, Luke!

**Pull request merged**
